src/screenshot_gui.cpp

```diff
@@ -72,3 +72,21 @@ void ShowScreenshotWindow()
 	CloseWindowById(WC_SCREENSHOT, 0);
 	new ScreenshotWindow(&_screenshot_window_desc);
 }
+
+/**
+ * Hides or shows the screenshot window when taking a screenshot.
+ * @param hidden Are we hiding the window or showing it again after the screenshot is taken?
+ */
+void SetScreenshotWindowHidden(bool hidden)
+{
+	ScreenshotWindow *scw = (ScreenshotWindow *)FindWindowById(WC_SCREENSHOT, 0);
+	if (scw != nullptr) {
+		if (hidden) {
+			scw->SetDirty();
+			SetBit(scw->left, 30); // Move window out of main screen
+		} else {
+			ClrBit(scw->left, 30); // Return window to original position
```

Comment:
Although I get it is easy and cheap to use setting a bit, it is confusing to the reader. Why bit 30? Why not 31? Why not 29? So many questions :)

It might be better to make a const with some large number, and add/subtract that value. It still is a hack, but I understand why you do it.

There is another approach to take, but I am not sure it is better/worse. As hide/show is always called in a pair, you could use a `static` value to safely store the current `scw->left`, and set the `scw->left` to a known off-screen value when hiding. That in effect gives the same result.

Basically, now a bit in `left` is used to indicate it is hidden or not, which is just a bit nasty. And might hurt us in future refactors. Personally, I would have gone for the `static` method, but I am not sure what other developers think about it. For sure the bitset is just asking for trouble, so a const to add/subtract to `scw->left` sounds more sane to me :)

Reply:
Oops, missed that @frosch123 also commented on this (but GitHub was hiding that comment from me). His solution would also work fine, as between the hide and show commands, the width of the screen cannot change. So that is also a safe thing to use in this case.